.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+information may be ignored or punted back to you, delaying a fix.  The below
+constitutes a bare minimum; more info is almost always better:
+ 
+* **What version of the core programming language interpreter/compiler are you
+ using?** For example, Most of our services are written in Java so you what version of Java you're on?
+* **What operating system are you on?** Windows? (10? Home? Pro?)
+ Mac OS X?  (10.x.x?) Linux? (Which distro? Which version of that
+ distro? 32 or 64 bits?) Again, more detail is better.
+* **Are you using latest codebase and are on main branch?** Ideally, you
+ followed the advice above and have ruled out (or verified that the problem
+ exists in) a few different versions.
+* **How can the developers recreate the bug on their end?** If possible,
+ include a copy of your code or an example repo, the command you used to invoke it, and the full
+ output of your run (if applicable.)
+ 
+### Version control branching
+-------------------------
+ 
+* Always **make a new branch** for your work, no matter how small. This makes
+ it easy for others to take just that one set of changes from your repository,
+ in case you have multiple unrelated changes floating around.
+ 
+   * A corollary: **don't submit unrelated changes in the same branch/pull
+     request**! The maintainer shouldn't have to reject your awesome bugfix
+     because the feature you put in with it needs more review.
+ 
+* **Base your new branch off of the appropriate branch** on the main
+ repository:
+ 
+   * **Bug fixes** should be based on the branch named after the **oldest
+     supported release line** the bug affects.
+ 
+       * E.g. if a feature was introduced in 1.1, the latest release line is
+         1.3, and a bug is found in that feature - make your branch based on
+         1.1.  The maintainer will then forward-port it to 1.3 and master.
+       * Bug fixes requiring large changes to the code or which have a chance
+         of being otherwise disruptive, may need to base off of **main**
+         instead. This is a judgement call -- ask the devs!
+ 
+   * **New features** should branch off of **the 'main' branch**.
+ 
+       * Note that depending on how long it takes for the dev team to merge
+         your patch, the copy of ``main`` you worked off of may get out of
+         date! If you find yourself 'bumping' a pull request that's been
+         sidelined for a while, **make sure you rebase or merge to latest
+         main** to ensure a speedier resolution.
+ 
+### Tests aren't optional
+---------------------
+ 
+Any bugfix that doesn't include a test proving the existence of the bug being
+fixed, may be suspect.  Ditto for new features that can't prove they actually
+work.
+ 
+We've found that test-first development really helps make features better
+architected and identifies potential edge cases earlier instead of later.
+Writing tests before the implementation is strongly encouraged.
+ 
+## Full example
```

Comment:
I am not sure I like this title.

e.g. Example workflow for submitting a change?